src/components/SettingsModal.js

```diff
@@ -0,0 +1,159 @@
+              render={({ field }) => (
+                <TextField
+                  label='Valor minimo para entrega grátis'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.minAmountForFreeDelivery ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.deliveryFee}
+              control={control}
+              rules={{ required: 'Valor de taxa entrega em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Valor de taxa entrega'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.deliveryFee ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              name={componentProps.name.discount}
+              control={control}
+              rules={{ required: 'Desconto em falta' }}
+              render={({ field }) => (
+                <TextField
+                  label='Percentagem de desconto'
+                  type={componentProps.type.text}
+                  variant={componentProps.variant.outlined}
+                  fullWidth
+                  sx={settingsClasses.formField}
+                  error={errors.discount ? true : false}
+                  autoComplete='true'
+                  {...field}
+                />
+              )}
+            />
+
+            <Controller
+              control={control}
+              name={componentProps.name.haveGlobalDeliveryDiscount}
+              render={({ field }) => <FormControlLabel control={<Switch checked={haveGlobalDeliveryDiscount} {...field} />} label='Entrega gratuita geral' />}
+            />
+
+            <button type={componentProps.type.submit} ref={submitFormRef} hidden>
+              Criar
+            </button>
+          </form>
+        </Box>
+        {errorMessage && <ErrorMessage message={errorMessage} />}
+        {errors.minAmountForFreeDelivery && <ErrorMessage message={errors.minAmountForFreeDelivery.message} />}
+        {errors.haveGlobalDeliveryDiscount && <ErrorMessage message={errors.haveGlobalDeliveryDiscount.message} />}
+        {errors.deliveryFee && <ErrorMessage message={errors.deliveryFee.message} />}
+        {errors.discount && <ErrorMessage message={errors.discount.message} />}
+        {!isLoading && (
+          <Button type={componentProps.type.button} variant={componentProps.variant.contained} onClick={() => submitFormRef.current.click()}>
```

Comment:
It's better if you use this button inside the form

Reply:
Yes i´m gonna do that. 😉